Migration of IDE Troubleshooting from Support Articles to central doc… (#15582)
rstudio/rstudio: docs/user/rstudio/ide/guide/troubleshooting/diagnostic-report.qmd

---

> pagetitle: "Creating a Diagnostic Report"

Reviewer: ```suggestion
title: "Creating a Diagnostic Report"
```

date-meta: 2025-01-07
---

> # Creating a diagnostic report

Reviewer: remove

Author: @AshleyHenry15 same as the previous question - the whole section, just line 3, just line 6, or some combination? Thanks

Diagnostic reports can be created within a working session, or from outside a session on a terminal.

## In an RStudio session

> Select from the menu **Help > Diagnostics > Write Diagnostics Report**

Reviewer: ```suggestion
From the menu , click **Help** > **Diagnostics** > **Write Diagnostics Report**.
```

## Outside of an RStudio session

> If RStudio is not starting, you can use the following method to run a diagnostics report. Note that you need to also show us the contents of the terminal session that is started.

Reviewer: ```suggestion
If RStudio is not starting, run a diagnostics report using the following method. Additionally, you need to provide the contents of the started terminal session.
```

Using the following method - add link to section

### Windows

> You can run a diagnostics report by typing the following command into **Start > Run**:

Reviewer: ```suggestion
Run a diagnostics report by typing the following command into **Start** > **Run**:
```

> ```powershell

Reviewer: idk if powershell is a supported input language here. I wouldn't use that

Author: I tried `cmd`, which is the old Windows default terminal type, and it isn't supported and the code looked bad. I tried `powershell` which is the new Windows terminal, and the commands looked much better. It is supported - you can see it here: https://github.com/jgm/skylighting/blob/master/skylighting-core/xml/powershell.xml

"C:\Program Files\RStudio\rstudio.exe" --run-diagnostics
```

> The diagnostics report will be placed in your user's Documents folder in a folder called `rstudio-diagnostics`, for example `C:\Users\Username\Documents\rstudio-diagnostics\diagnostics-report.txt`

Reviewer: ```suggestion
The diagnostics report is placed in your user's Documents directory in a folder named `rstudio-diagnostics`. For example: `C:\Users\Username\Documents\rstudio-diagnostics\diagnostics-report.txt`
```

> Note If you installed RStudio to a different location than `C:\Program Files` make sure to specify this location in the command.

Reviewer: Don't use "Note'

```suggestion
If you installed RStudio to a location other than `C:\Program Files`, specify this unique location in the command.
```

> ### Mac OS X

Reviewer: I don't think that this is correct anymore?

Isn't Mac OS X now macOS?

> You can run a diagnostics report by typing the following command at the Terminal:

Reviewer: ```suggestion
Run a diagnostics report by typing the following command at the Terminal:
```

> ```zsh

Reviewer: IDK if zsh is supported?

Author: Yes, zsh is supported. I've tested them all - `quarto preview` with and without.

/Applications/RStudio.app/Contents/MacOS/RStudio --run-diagnostics
```

> The diagnostics report will be placed in your user home directory, for example `~/rstudio-diagnostics/diagnostics-report.txt`.

Reviewer: ```suggestion
The diagnostics report is placed in your user home directory. For example: `~/rstudio-diagnostics/diagnostics-report.txt`
```

### Linux

> You can run a diagnostics report by typing the following command at the Terminal:

Reviewer: ```suggestion
Run a diagnostics report by typing the following command at the Terminal:
```

> ```bash

Reviewer: I think that bash is supported? We need to check on these.

Author: 👍🏻

rstudio --run-diagnostics
```

> The diagnostics report will be placed in your user home directory, for example `~/rstudio-diagnostics/diagnostics-report.txt`.

Reviewer: ```suggestion
The diagnostics report is placed in your user home directory. For example: `~/rstudio-diagnostics/diagnostics-report.txt`
```
